pkg/core/interfaces.go

```diff
@@ -171,3 +147,30 @@ type Requests interface {
 	MakeAPIRequest(ctx context.Context, httpMethod, endpoint string, body []byte, params,
 		headers map[string]string) (rawbody []byte, statusCode int, err error)
 }
+
+// ListSubModuleService will sends the submodule count to TAS server
+type ListSubModuleService interface {
+	// Send sends count of submodules to TAS server
+	Send(ctx context.Context, buildID string, totalSubmodule int) error
+}
+
+// Driver has the responsibility to run discovery and test execution
+type Driver interface {
+	// RunDiscovery runs the test discovery
+	RunDiscovery(ctx context.Context, payload *Payload,
+		taskPayload *TaskPayload, oauth *Oauth, coverageDir string, secretMap map[string]string) error
+	// RunExecution runs the test execution
+	RunExecution(ctx context.Context, payload *Payload,
+		taskPayload *TaskPayload, oauth *Oauth, coverageDir string, secretMap map[string]string) error
+}
+
+// LogWriterStartegy will be implemented by all different type of log writer
```

Comment:
this comment is not very useful. Since, it is an interface, it inherently means it needs an implementation.